File: Shared/src/main/java/com/lenta/shared/models/core/ExciseStamp.java
package com.lenta.shared.models.core;

import lombok.Getter;

public class ExciseStamp implements IExciseStamp {

    @Getter private String materialNumber;
    @Getter private String code;

    public ExciseStamp(String materialNumber, String code)
    {
        this.materialNumber = materialNumber;
        this.code = code;
    }

    @Override
    public int egaisVersion() {
        return getEgaisVersion(code);
    }

    public static int getEgaisVersion(String code)
    {
        switch (code.length())
        {
            case EgaisStampVersion.V2:
                return EgaisStampVersion.V2;
            case EgaisStampVersion.V3:
                return EgaisStampVersion.V3;
            default:
                return EgaisStampVersion.UNKNOWN;
        }
    }

}

File: Shared/src/main/java/com/lenta/shared/models/core/ExciseStamp.kt
package com.lenta.shared.models.core

open class ExciseStamp(val materialNumber: String, val code: String) : IExciseStamp {
    override fun egaisVersion(): Int {
        return getEgaisVersion(code)
    }

    companion object {
        fun getEgaisVersion(code: String): Int {
            return when (code.length) {
                // Все марки определяем как 150
                EgaisStampVersion.V2.version,
                EgaisStampVersion.V3.version -> EgaisStampVersion.V3.version
                else -> EgaisStampVersion.UNKNOWN.version
            }
        }
    }
}